x-pack/test/plugin_functional/plugins/resolver_test/public/plugin.ts

```diff
@@ -23,43 +23,35 @@ export class ResolverTestPlugin
       ResolverTestPluginSetupDependencies,
       ResolverTestPluginStartDependencies
     > {
-  private resolveEmbeddable!: (
-    value: IEmbeddable | undefined | PromiseLike<IEmbeddable | undefined> | undefined
-  ) => void;
-  /**
-   * We register our application during the `setup` phase, but the embeddable
-   * plugin API is not available until the `start` phase. In order to access
-   * the embeddable API from our application, we pass a Promise to the application
-   * which we resolve during the `start` phase.
-   */
-  private embeddablePromise: Promise<IEmbeddable | undefined> = new Promise<
-    IEmbeddable | undefined
-  >(resolve => {
-    this.resolveEmbeddable = resolve;
-  });
-  public setup(core: CoreSetup) {
+  public setup(core: CoreSetup<ResolverTestPluginStartDependencies>) {
     core.application.register({
       id: 'resolver_test',
       title: i18n.translate('xpack.resolver_test.pluginTitle', {
         defaultMessage: 'Resolver Test',
       }),
       mount: async (_context, params) => {
+        let resolveEmbeddable: (
```

Comment:
I'm assuming you are still using Promises (inside of this await) because `renderApp()` has logic to "listen" to it, right?
Would it be simpler to chain off the `getStartServices()` and pass that to `renderApp()`?

```typescript
const resolveEmbeddable = core.getStartServices().then(([, { embeddable}]) => {
    const factory = embeddable.getEmbeddableFactory('resolver');
    return Promise.resolve(factory.create({ id: 'test basic render' }));
});
//...
return renderApp(params, resolveEmbeddable);
```

Reply:
@paul-tavares possibly. The email from @joshdover does that. However that would block the app from rendering until the promise resolves. If the promise resolves in the same tick then I suppose there's no harm. 

/shrug